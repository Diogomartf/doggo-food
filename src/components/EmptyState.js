import React from 'react';
import styled from 'styled-components';
import Dog from '../images/dog.svg';

const EmptyState = () => (
  <Flex>
    <Image src={Dog} alt="Dog with food" />
    <Title>No results</Title>
    <Text>Search for ingredients above!</Text>
  </Flex>
);

const Flex = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

const Image = styled.img`
  width: 100%;
  height: 100%;
  max-height: 360px;
  margin-top: 32px;
`;

const Title = styled.div`
  margin-top: 12px;
  margin-bottom: 4px;
  text-align: center;

  font-size: 26px;
  font-weight: 600;
  color: ${({ theme }) => theme.colors.black};
`;

const Text = styled.div`
  margin-top: 8px;
  font-size: 18px;
  color: ${({ theme }) => theme.colors.darkGray};
`;

EmptyState.displayName = 'EmptyState';

export default EmptyState;
