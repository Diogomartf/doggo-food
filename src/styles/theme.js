import colors from './colors';

export default {
  name: 'default theme',
  colors
};
