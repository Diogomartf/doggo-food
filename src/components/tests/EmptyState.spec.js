import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../../styles';

import EmptyState from '../EmptyState';

describe('<EmptyState/>', () => {
  it('should render correctly', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <EmptyState />
      </ThemeProvider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
