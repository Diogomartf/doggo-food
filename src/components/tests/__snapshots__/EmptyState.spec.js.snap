// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyState/> should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#2B2B2B",
        "darkGray": "#626262",
        "gray": "#EAEAEA",
        "lightGray": "#F4F4F4",
        "primary": "#6056F5",
        "red": "#CF0A2C",
        "white": "#FFFFFF",
        "yellow": "#FBB03B",
      },
      "name": "default theme",
    }
  }
>
  <EmptyState>
    <styled.div>
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "bxcuYP",
              "rules": Array [
                "
  display: flex;
  flex-direction: column;
  align-items: center;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
      >
        <div
          className="sc-bdVaJa bxcuYP"
        >
          <styled.img
            alt="Dog with food"
            src="dog.svg"
          >
            <StyledComponent
              alt="Dog with food"
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "lastClassName": "kXEWvP",
                    "rules": Array [
                      "
  width: 100%;
  height: 100%;
  max-height: 360px;
  margin-top: 32px;
",
                    ],
                  },
                  "displayName": "styled.img",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": "img",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              src="dog.svg"
            >
              <img
                alt="Dog with food"
                className="sc-bwzfXH kXEWvP"
                src="dog.svg"
              />
            </StyledComponent>
          </styled.img>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-htpNat",
                    "isStatic": false,
                    "lastClassName": "iJCCbU",
                    "rules": Array [
                      "
  margin-top: 12px;
  margin-bottom: 4px;
  text-align: center;

  font-size: 26px;
  font-weight: 600;
  color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-htpNat",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-htpNat iJCCbU"
              >
                No results
              </div>
            </StyledComponent>
          </styled.div>
          <styled.div>
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bxivhb",
                    "isStatic": false,
                    "lastClassName": "egWtGW",
                    "rules": Array [
                      "
  margin-top: 8px;
  font-size: 18px;
  color: ",
                      [Function],
                      ";
",
                    ],
                  },
                  "displayName": "styled.div",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bxivhb",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
            >
              <div
                className="sc-bxivhb egWtGW"
              >
                Search for ingredients above!
              </div>
            </StyledComponent>
          </styled.div>
        </div>
      </StyledComponent>
    </styled.div>
  </EmptyState>
</ThemeProvider>
`;
